Presentation.Web/app/components/data-processing/tabs/data-processing-agreement-tab-roles.controller.ts

```diff
@@ -0,0 +1,271 @@
+            "select2LoadingService"
+        ];
+
+        constructor(
+            private readonly dataProcessingAgreementService: Services.DataProcessing.IDataProcessingAgreementService,
+            private readonly notify,
+            public hasWriteAccess,
+            private readonly dataProcessingAgreement: Models.DataProcessing.IDataProcessingAgreementDTO,
+            private readonly dataProcessingAgreementRoles: Models.DataProcessing.IDataProcessingRoleDTO[],
+            private readonly select2LoadingService: Services.ISelect2LoadingService) {
+        }
+
+        headerName = this.dataProcessingAgreement.name;
+
+        roles = _.map(this.dataProcessingAgreementRoles, role => this.mapToRoleOptions(role));
+
+        selectedRoleIdAsString = this.roles[0].id.toString();
+        selectedUser: Models.ViewModel.Generic.Select2OptionViewModel;
+
+        userOptions = this.getAvailableUserOptions(() => parseInt(this.selectedRoleIdAsString));
+
+        assignedRoles = _.orderBy(
+            _.map(this.dataProcessingAgreement.assignedRoles, assignedRole => this.mapToEditableAssignedRole(assignedRole)),
+            assignedRole => [assignedRole.role.name, assignedRole.user.name, assignedRole.user.email],
+            "asc");
+
+        lastSortedBy = "";
+        shouldSortAsc = true;
+        sortAssignedRoles(sortBy: string) {
+            sortBy === this.lastSortedBy ? this.shouldSortAsc = !this.shouldSortAsc : this.shouldSortAsc = true;
+            var sortDirection = this.shouldSortAsc ? "asc" : "desc";
+            switch (sortBy) {
+                case "roleName":
+                    this.assignedRoles = _.orderBy(
+                        this.assignedRoles,
+                        assignedRole => [assignedRole.role.name, assignedRole.user.name, assignedRole.user.email],
+                        sortDirection);
+                    break;
+                case "userName":
+                    this.assignedRoles = _.orderBy(
+                        this.assignedRoles,
+                        assignedRole => [assignedRole.user.name, assignedRole.user.email, assignedRole.role.name],
+                        sortDirection);
+                    break;
+                case "userEmail":
+                    this.assignedRoles = _.orderBy(
+                        this.assignedRoles,
+                        assignedRole => [assignedRole.user.email, assignedRole.user.name, assignedRole.role.name],
+                        sortDirection);
+                    break;
+            }
+            this.lastSortedBy = sortBy;
+        }
+
+        submitRole() {
+            if (this.selectedUser === null || angular.isUndefined(this.selectedUser)) {
+                return;
+            }
+            if (this.selectedUser.id === null || angular.isUndefined(this.selectedUser.id)) {
+                return;
+            }
+
+            var selectedRoleId = parseInt(this.selectedRoleIdAsString);
```

Comment:
i stedet for parseInt(..) alle steder hvor du bruger den så lav en hjælpemetode.. getSelectedRoleId()